test/e2e/karmadactl_test.go

```diff
@@ -1154,6 +1154,87 @@ var _ = ginkgo.Describe("Karmadactl options testing", func() {
 	})
 })
 
+var _ = ginkgo.Describe("Karmadactl taint testing", func() {
+	ginkgo.Context("Test karmadactl taint command with different effects", func() {
+		var clusterName string
+		var taintKey = "environment"
+		var taintValue = "production"
+
+		ginkgo.BeforeEach(func() {
+			clusterName = framework.ClusterNames()[0]
+		})
+
+		// testTaintEffect is a reusable function for testing taints on a given effect.
+		testTaintEffect := func(effect corev1.TaintEffect) {
+			ginkgo.It(fmt.Sprintf("should handle %s taints correctly", effect), func() {
+				taint := fmt.Sprintf("%s=%s:%s", taintKey, taintValue, effect)
+				taintRemove := fmt.Sprintf("%s-", taint)
+
+				ginkgo.By(fmt.Sprintf("Applying %s taint to the cluster", effect), func() {
+					cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "taint", "clusters", clusterName, taint)
+					output, err := cmd.ExecOrDie()
+					gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+					gomega.Expect(strings.Contains(output, fmt.Sprintf("cluster/%s tainted", clusterName))).Should(gomega.BeTrue())
```

Comment:
```suggestion
					gomega.Expect(strings.Contains(output, fmt.Sprintf("cluster/%s tainted", clusterName))).Should(gomega.BeTrue())
					framework.WaitClusterFitWith(controlPlaneClient, clusterName, func(cluster *clusterv1alpha1.Cluster) bool {
						for _, taint := range cluster.Spec.Taints {
							if taint.Key == taintKey && taint.Value == taintValue && taint.Effect == effect {
								return true
							}
						}
						return false
					})
```

Add a check to determine whether the cluster's taint field meets expectations would be better~